javascript/javascript1/week2/answer.js

```diff
@@ -0,0 +1,146 @@
+if so then checks if the person is a woman and returns the string accordingly
+*/
+function getFullname(firstname, surname, useFormalName, isWoman) {
+  if (useFormalName) {
+    if (isWoman) {
+      return "Madam " + firstname + " " + surname;
+    } else {
+      return "Lord " + firstname + " " + surname;
+    }
+  } else {
+    return firstname + " " + surname;
+  }
+}
+
+const fullname1 = "Benjamin";
+const fullname2 = "Hughes";
+const useFormal = true;
+const isFemale = true;
+const display = getFullname(fullname1, fullname2, useFormal, isFemale);
+console.log(display);
+
+//Event application
+function getEventWeekday(daysFromToday) {
+  let dateObject = new Date();
+  let dayNumber = dateObject.getDay(); //returns a number from 0-6
+  let dayNames = [
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+  const indexes = daysFromToday + dayNumber;
+  if (indexes >= 7) {
+    let k = indexes % 7; //since dayNames has only 7 elements need to find the remainder
+    return dayNames[k];
+  } else {
+    return dayNames[indexes];
+  }
+}
+console.log(getEventWeekday(10));
+
+//Weather wear
+function findTypeOfClothes(temperature) {
+  if (temperature < 0) {
+    return "multiple layers of coat and jacket";
+  } else if (temperature >= 0 && temperature <= 10) {
+    return "sweaters and long pants";
+  } else if (temperature > 10 && temperature < 18) {
+    return "light coats or cotton clothes";
+  } else {
+    return "shorts and a t-shirt";
+  }
+}
+const clothesToWear = findTypeOfClothes(18);
+console.log(`You can wear ${clothesToWear}`);
+
+//Student Manager
+const class07Students = [];
+function addStudentToClass(studentName) {
+  if (class07Students.length < 6 || studentName === "queen") {
+    if (class07Students.includes(studentName)) {
+      console.log(`Student ${studentName} is already in the class`);
+    } else {
+      class07Students.push(studentName);
+    }
+  } else {
+    console.log("Cannot add more students to class 07");
+  }
+}
+
+function getNumberOfStudents() {
+  return class07Students.length;
+}
+
+//trying out different cases as mentioned
+addStudentToClass("krishna");
+addStudentToClass("glen");
+addStudentToClass("dan");
+addStudentToClass("alex");
+addStudentToClass("benjamin");
+addStudentToClass("glen");
+addStudentToClass("hughes");
+addStudentToClass("queen");
+addStudentToClass("queen");
+addStudentToClass("joe");
+
+const totalStudents = getNumberOfStudents();
+console.log(`Number of Students in class07 : ${totalStudents}`);
+
+//Candy Helper
+//created an array of objects to store different types of candies and their prices
+const candies = [
+  {
+    type: "Sweet",
+    pricePerGram: 0.5,
+  },
+  {
+    type: "Chocolate",
+    pricePerGram: 0.7,
+  },
+  {
+    type: "Toffee",
+    pricePerGram: 1.1,
+  },
+  {
+    type: "Chewing-gum",
+    pricePerGram: 0.03,
+  },
+];
+
+const boughtCandyPrices = [];
```

Comment:
Thank you for doing this optional exercise. It's the hardest for that week and you've done great. Congratulations on the effort and dedication!